src/app/reducers/query.ts

```diff
@@ -0,0 +1,150 @@
+import { createSelector } from 'reselect';
+
+import * as search from '../actions/query';
+import { Query, ReloactionAfterQuery, fromRegExp, followersRegExp,
+					FilterList, TimeBound  } from '../models';
+
+/**
+ * Each reducer module must import the local `State` which it controls.
+ *
+ * Here the `State` contains two properties:
+ * @prop [query: Query] Query object on which the search is based.
+ *
+ */
+export interface State {
+	query: Query;
+}
+
+/**
+ * Initial Query state which is passed onto the store.
+ */
+const queryInitialState: Query =  {
+	displayString: '',
+	queryString: '',
+	location: ReloactionAfterQuery.NONE,
+	filter: {
+		audio: false,
+		video: false,
+		images: false
+	},
+	near: null,
+	timeBound: {
+		since: null,
+		until: null
+	},
+	from: false,
+	followers: false
+};
+
+/**
+ * There is always a need of initial state to be passed onto the store.
+ *
+ * @prop: query: ''
+ * @prop: loading: false
+ */
+export const initialState: State = {
+	query: queryInitialState,
+};
+
+
+/**
+ * The actual reducer function. Reducers can be thought of as the tables in the DataBase.
+ * These are the functions which are responsible for maintaing, and updating the
+ * `State` of the application they control.
+ *
+ * Here the reducer cotrols that part of the state which is shows the state of the application
+ * wheather it is searching and what is it searching for.
+ */
+export function reducer(state: State = initialState, action: search.Actions): State {
+	switch (action.type) {
+		case search.ActionTypes.VALUE_CHANGE: {
+			const query: any = action.payload;
+			const isFromQuery: boolean = (fromRegExp.exec(query)) ? true : false;
+			const isFollowerQuery: boolean = (followersRegExp.exec(query)) ? true : false;
+
+			return Object.assign({}, state, {
+				query: {
+					...state.query,
+					displayString: query,
+					from: isFromQuery,
+					follower: isFollowerQuery
+				},
+				showUserFeed: isFollowerQuery
```

Comment:
Yeah will remove it, this was really not a decent way to handle the `followers:<>` query, what was implemented earlier. I have something else in mind, will share  when this PR is done, don't want to make too many changes now, already there are 30+ modified files :/